src/lazy/text/buffer.rs

```diff
@@ -2709,6 +2762,56 @@ mod tests {
             mismatch_sexp(input);
         }
     }
+    #[test]
+    fn test_match_sexp_1_1() {
```

Comment:
🗺️ I added some tests for Ion 1.1 SExps and Lists, as well as some test cases for E-Expressions that are nested inside other things. This is far from comprehensive, but it's at least an improvement.